### PR TITLE
update: One buffer for input and one for output

### PR DESCRIPTION
**Pull request by vargac**
* vic_buffer renamed to vic_buffer_in
* Added vic_buffer_out for outputing vic functions
* Saving stack
* Reduced input buffer size, there is no command for vic with such length

Signed-off-by: vargac <simon.varga123@gmail.com>

### CODE REVIEW

File: src/vic.h [hunk 1]
```diff
@@ -74,12 +75,13 @@
...
-extern uint8_t vic_buffer_len;
+extern uint8_t vic_buffer_in_len;
+extern char vic_buffer_in[VIC_BUFFER_IN_SIZE + 1];
+extern char vic_buffer_out[VIC_BUFFER_OUT_SIZE + 1];
 
-#define vic_printf(format, ...) do { char _vic_buff[VIC_BUFFER_SIZE];\
-    snprintf(_vic_buff, VIC_BUFFER_SIZE, (format), __VA_ARGS__);\
-    vic_print(_vic_buff); } while(0)
+#define vic_printf(format, ...) do {\
+    snprintf(vic_buffer_out, VIC_BUFFER_OUT_SIZE + 1, (format), __VA_ARGS__);\
```
Comment (mrshu) [inline src/vic.h]: @vargac if I read this correct, whatever is in `vic_buffer_out` will be replaced here. Are we OK with that?

File: src/vic.h [hunk 2]
```diff
@@ -52,7 +52,8 @@
 
 #endif
 
-#define VIC_BUFFER_SIZE 128
+#define VIC_BUFFER_IN_SIZE 64
```
Comment (mrshu) [inline src/vic.h]: @vargac so this makes the maximum length of a command to be `64`, right?

**Comment by mrshu** [comment]
@Adman would you mind taking a look?

File: src/vic.h (hunk 2 above)
Comment (vargac) [inline src/vic.h]: @mrshu Yes, the longest command is "set 'var name' 'var value'", that makes 30 bytes, but there may be more spaces, so I've chosen 64.

File: src/vic.h (hunk 1 above)
Comment (vargac) [inline src/vic.h]: @mrshu vic_buffer_out is always used only for a moment and then is printed, so that doesn't make problem

**Comment by Adman** [comment]
@mrshu I'm not gonna review this as I do not have time for reading entire code of this project

**Comment by mrshu** [comment]
Overall this makes a lot of sense so I'll just merge it.

**Comment by mrshu** [comment]
Nice job @vargac btw!

**Pull request merged**